### ISSUE TITLE
[a11y][Violation][4.1.2 Name, Role, Value] Accessibility issues reported on JupyterLab UI using IBM Equal Access Accessibility Checker tool

### ISSUE THREAD

**Issue by nkn2022** [jupyterlab/jupyterlab#16890]
## Description

I am running an accessibility test suite with Jupyter using the **IBM Equal Access Accessibility Checker** tool on Chrome browser. The results look good overall, only a few things are missing to reach a first formal level of compliance. So kindly help look into these. 

Issues reported were violating **WCAG 2.2** requirements. 
Ref: https://www.ibm.com/able/requirements/checker-rule-sets

Listing of violations categorized based on the checkpoints. 
Full report (includes information on element locations where issue were found): 
[Accessibility_Report-Untitled.ipynb - JupyterLab.xlsx](https://github.com/user-attachments/files/17383160/Accessibility_Report-Untitled.ipynb.-.JupyterLab.xlsx)

# Category: `4.1.2 Name, Role, Value`
Ref: https://www.ibm.com/able/requirements/requirements/?version=v7_3#4_1_2

****Violation of Rule:** `aria_attribute_required`**

> List of issues with elements that violate this rule with a screenshots
> > **Issue 1** 
> > Element with 'switch' role does not have the required ARIA attribute(s): 'aria-checked'
> > **Element location:** 
> > `<button title="Simple Interface (⇧ ⌘ D)" role="switch" class="jp-switch">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/0d33bd73-a40e-4f24-8b47-b7b2f64ce7fc">

****Violation of Rule:** `element_tabbable_role_valid`**

> List of issues with elements that violate this rule with a screenshots
> > **Issue 1** 
> > The tabbable element's role 'none' is not a widget role
> > **Element location:** 
> > `<jp-button minimal="" appearance="stealth" current-value="" tabindex="0" title="New Launcher (⇧ ⌘ L)" data-command="launcher:create" aria-label="New Launcher (⇧ ⌘ L)" aria-disabled="false" class="jp-ToolbarButtonComponent">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/5d8350aa-b485-4980-9820-34c7204eb9ab">

> > **Issue 2**
> > The tabbable element's role 'none' is not a widget role
> > **Element location:** 
> > `<span tabindex="0" class="jp-DirListing-itemText">`
> > **Screenshot:**
> ><img width="468" alt="image" src="https://github.com/user-attachments/assets/7a6f4999-de2c-4344-afa1-4bc1f7aaf3c3">

> > **Issue 3**
> > The tabbable element's role 'none' is not a widget role
> > **Element location:** 
> > `<jp-button minimal="" appearance="stealth" current-value="" tabindex="0" title="Insert a cell below (B)" data-command="notebook:insert-cell-below" aria-label="Insert a cell below (B)" aria-disabled="false" class="jp-ToolbarButtonComponent">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/7258415e-f25f-4142-accc-ca81b9ce1238">

> > **Issue 4**
> > The tabbable element's role 'none' is not a widget role
> > **Element location:** 
> > `<div data-windowed-list-index="0" tabindex="0" aria-label="Code Cell Content with Output" class="lm-Widget jp-Cell jp-CodeCell jp-Notebook-cell jp-mod-selected jp-mod-active">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/0e27e0dd-537c-48e7-a0ee-43df8230505d">

> > **Issue 5**
> > The tabbable element's role 'none' is not a widget role
> > **Element location:** 
> > `<jp-button minimal="" appearance="stealth" current-value="" tabindex="0" title="Create a duplicate of this cell below" data-command="notebook:duplicate-below" aria-label="Create a duplicate of this cell below" aria-disabled="false" class="jp-ToolbarButtonComponent">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/e1416a4a-a295-48d1-bb14-2a0302bba99a">

> > **Issue 6**
> > The tabbable element's role 'none' is not a widget role
> > **Element location:** 
> > `<div tabindex="0" class="jp-StatusBar-GroupItem ">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/a850d84f-7f62-47fb-8bd0-db468643ed81">

> > **Issue 7**
> > The tabbable element's role 'none' is not a widget role
> > **Element location:** 
> > `<span tabindex="0" title="Change kernel for Untitled.ipynb" class="jp-StatusBar-TextItem undefined">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/f5a5f672-0bca-4a98-be9a-f6ba63e57181">

> > **Issue 8**
> > The tabbable element's role 'none' is not a widget role
> > **Element location:** 
> > `<span tabindex="0" title="Go to line number…" class="jp-StatusBar-TextItem undefined">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/f106dee7-b89b-437a-9292-4a32cdacebdc">

****Violation of Rule:** `input_label_exists`**

> List of issues with elements that violate this rule with a screenshots
> > **Issue 1** 
> > Form control element <button> has no associated label
> > **Element location:** 
> > `<button class="clear-button clear-button__hidden" tabindex="-1" part="clear-button">`
> > **Screenshot:**
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/126b7740-1f76-4370-9423-b3efd68ad8cd">

> > **Issue 2** 
> > Form control with "textbox" role has no associated label
> > **Element location:** 
> > `<div data-language="python" style="tab-size: 4;" tabindex="-1" aria-multiline="true" role="textbox" class="cm-content" contenteditable="true" translate="no" autocapitalize="off" autocorrect="off" spellcheck="false">`
> > **Screenshot 1:**
> >  <img width="468" alt="image" src="https://github.com/user-attachments/assets/31a751b2-1ff6-4905-8df2-61f751d09349">
> > **Screenshot 2:** 
> > <img width="468" alt="image" src="https://github.com/user-attachments/assets/38207e49-6d89-4702-9fc6-bdaa62f0d3e6">
> > **Screenshot 3:**
> >  <img width="468" alt="image" src="https://github.com/user-attachments/assets/a6aeaccd-6003-4d39-b7a4-cfac4712369c">

## Reproduce

- Install Chrome extension for IBM accessibility checker tool. Ref: https://www.ibm.com/able/toolkit/verify/automated
- Open JupyterLab UI on Chrome browser. Open `notebook` file to edit.
- Run the tool and we can see the above `Violation`s reported.

Here is the screenshot of the UI scanned for the attached report:

<img width="1534" alt="image" src="https://github.com/user-attachments/assets/0e0cbc5c-cf21-4393-97fe-b566da24b58b">

## Expected behavior

No violations are reported on the scanning to reach a first formal level of compliance.

## Context

- Operating System and version: MacOS Sonoma Version 14.5
- Browser and version: Chrome Version 129+
- JupyterLab version: Version 4.2.5

**Comment by JasonWeill** [comment]
See also #4878, concerning usability, which mentions ARIA roles.

**Comment by krassowski** [comment]
> Is there an alternate way to achieve focus on this component for people who are using keyboard?

Isn't it just pressing Enter when the cell is focused?

**Comment by nkn2022** [comment]
https://github.com/jupyterlab/jupyterlab/issues/17680 to track a tab order issue on the status bar.

**Comment by nkn2022** [comment]
Re-tested on Jupyterlab  v4.4.4
Some issues are fixed. So I will update the description of this issue to reflect this. 

Remaining issues: 
(Mostly `jp-button` issues which needs to be addressed based on the [discussion](https://github.com/jupyterlab/jupyterlab/pull/17591#discussion_r2107045015) in the [PR](https://github.com/jupyterlab/jupyterlab/pull/17591))

****Violation of Rule:** `element_tabbable_role_valid`** 
> > **Issue 1** 
> > **Issue 3** 
> > **Issue 4** (notebook)
> > **Issue 5** (notebook)
> > **Issue 9** 

****Violation of Rule:** `input_label_exists`**
> > **Issue 1** 
> > **Issue 2**  (notebook)

**Comment by nkn2022** [comment]
Re-tested on Jupyterlab v4.4.9

Remaining issues: 
(`jp-button` issues which needs to be addressed based on the [discussion](https://github.com/jupyterlab/jupyterlab/pull/17591#discussion_r2107045015) in the [PR](https://github.com/jupyterlab/jupyterlab/pull/17591))

****Violation of Rule:** `element_tabbable_role_valid`** 
> > **Issue 1** 
> > **Issue 3** 
> > **Issue 4** (notebook)
> > **Issue 5** (notebook)

****Violation of Rule:** `input_label_exists`**
> > **Issue 1** 
> > **Issue 2**  (notebook)
